contracts/modules/loan/BaseLoan.sol

```diff
@@ -259,14 +259,15 @@ abstract contract BaseLoan is ILoan, MutualUpgrade {
     uint256 totalOwed = debt.principal + debt.interestAccrued;
     require(totalOwed == _getMaxRepayableAmount(positionId, totalOwed));
 
-    // borrwer deposits remaining balance not already repaid and held in contract
+    // borrower deposits remaining balance not already repaid and held in contract
     bool success = IERC20(debt.token).transferFrom(
       msg.sender,
       address(this),
       totalOwed
     );
     require(success, 'Loan: deposit failed');
-
+    debts[positionId].principal = 0;
```

Comment:
So the logic to set this to 0 should be in _repay but i did remove _repay logic from BaseLoan because we needed 2 different things for Revolver vs Term.

I think its fine to remove _repay call on line 271 since we don't need all the extra logic inside it since we know we're paying 100% of the loan. And then can you emit events for RepayInterest and RepayPrincipal (done in _repay usually) before u delete the values and its good to do.